### PR TITLE
 Add DHL Shipper and DHL integration - DE -> AT shipping

### PR DESCRIPTION
**Pull request by TheBestPessimist**
https://github.com/metasfresh/metasfresh/issues/5611

### CODE REVIEW

File: de.metas.business/src/main/java/de/metas/customs/CustomsInvoiceRepository.java [hunk 1]
```diff
@@ -145,6 +150,35 @@ I_C_Customs_Invoice save(@NonNull final CustomsInvoice customsInvoice)
 		return customsInvoiceRecord;
 	}
 
+	public List<CustomsInvoiceLine> retrieveLines(@NonNull final CustomsInvoiceId customsInvoiceId)
```
Comment (metas-ts) [inline de.metas.business/src/main/java/de/metas/customs/CustomsInvoiceRepository.java]: why not get the customs invoice (entity root) here?
@teosarca wdyt?

File: de.metas.shipper.gateway.dhl/src/main/java/de/metas/shipper/gateway/dhl/DhlDeliveryOrderRepository.java [hunk 2]
```diff
@@ -122,12 +128,35 @@ private DeliveryOrder toDeliveryOrderFromPO(final I_DHL_ShipmentOrderRequest req
...
 
 		final ImmutableList<DhlCustomDeliveryDataDetail> dhlCustomDeliveryDataDetail = ordersPo.stream()
-				.map(po -> DhlCustomDeliveryDataDetail.builder()
-						.packageId(po.getPackageId())
-						.awb(po.getawb())
-						.sequenceNumber(DhlSequenceNumber.of(po.getDHL_ShipmentOrder_ID()))
-						.pdfLabelData(po.getPdfLabelData())
-						.build())
+				.map(po -> {
+					//					DhlCustomsDocument customsDocument = null;
```
Comment (metas-ts) [inline de.metas.shipper.gateway.dhl/src/main/java/de/metas/shipper/gateway/dhl/DhlDeliveryOrderRepository.java]: please clean this up, or add a comment explaining the use of this commented out code

File: de.metas.shipper.gateway.dhl/src/main/java/de/metas/shipper/gateway/dhl/DhlDeliveryOrderRepository.java [hunk 3]
```diff
@@ -313,6 +342,33 @@ private I_DHL_ShipmentOrderRequest toCreateShipmentOrderRequestPO(@NonNull final
 					shipmentOrder.setDHL_Shipper_CountryISO3Code(pickupAddress.getCountry().getAlpha3());
 				}
 
+				//				{
```
Comment (metas-ts) [inline de.metas.shipper.gateway.dhl/src/main/java/de/metas/shipper/gateway/dhl/DhlDeliveryOrderRepository.java]: again, please clean this up, or add a comment explaining the use of this commented out code

File: de.metas.shipper.gateway.dhl/src/main/java/de/metas/shipper/gateway/dhl/DhlShipperGatewayClient.java [hunk 4]
```diff
@@ -367,11 +369,49 @@ private CreateShipmentOrderRequest createDHLShipmentOrderRequest(
 					shipmentOrderTypeShipment.setShipper(shipperType);
 				}
 
+				//				{
```
Comment (metas-ts) [inline de.metas.shipper.gateway.dhl/src/main/java/de/metas/shipper/gateway/dhl/DhlShipperGatewayClient.java]: again, please clean this up, or add a comment explaining the use of this commented out code

File: de.metas.shipper.gateway.dhl/src/main/java/de/metas/shipper/gateway/dhl/model/DhlCustomDeliveryDataDetail.java [hunk 5]
```diff
@@ -46,13 +46,21 @@
...
 	@Builder(toBuilder = true)
 	private DhlCustomDeliveryDataDetail(
 			final int packageId,
 			@NonNull final DhlSequenceNumber sequenceNumber,
 			@Nullable final byte[] pdfLabelData,
 			@Nullable final String awb,
-			@Nullable final String trackingUrl)
+			@Nullable final String trackingUrl,
+			final boolean internationalDelivery
+			//			@Nullable final DhlCustomsDocument customsDocument
```
Comment (metas-ts) [inline de.metas.shipper.gateway.dhl/src/main/java/de/metas/shipper/gateway/dhl/model/DhlCustomDeliveryDataDetail.java]: agqain, please clean this up, or add a comment explaining the use of this commented out code

File: de.metas.shipper.gateway.dhl/src/main/java/de/metas/shipper/gateway/dhl/model/DhlCustomDeliveryDataDetail.java [hunk 6]
```diff
@@ -61,5 +69,13 @@ private DhlCustomDeliveryDataDetail(
 		this.pdfLabelData = pdfLabelData;
 		this.awb = awb;
 		this.trackingUrl = trackingUrl;
+		this.internationalDelivery = internationalDelivery;
+//		if (internationalDelivery && customsDocument == null)
```
Comment (metas-ts) [inline de.metas.shipper.gateway.dhl/src/main/java/de/metas/shipper/gateway/dhl/model/DhlCustomDeliveryDataDetail.java]: again, please clean this up, or add a comment explaining the use of this commented out code

File: de.metas.shipper.gateway.dhl/src/main/java/de/metas/shipper/gateway/dhl/model/DhlCustomsDocument.java [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+	@NonNull
+	BigDecimal additionalFee; // not sure which column inside customs invoice refers to this
+
+	@NonNull
+	String electronicExportNotification; // no clue, but i guess it should always have value "1"
+
+	@NonNull
+	String packageDescription;
+	//	@NonNull
+	//	String countryCode2Origin; // this is the source bpartner country
```
Comment (metas-ts) [inline de.metas.shipper.gateway.dhl/src/main/java/de/metas/shipper/gateway/dhl/model/DhlCustomsDocument.java]: again, please clean this up, or add a comment explaining the use of this commented out code

File: de.metas.business/src/main/java/de/metas/customs/CustomsInvoiceRepository.java (hunk 1 above)
Comment (TheBestPessimist) [reply de.metas.business/src/main/java/de/metas/customs/CustomsInvoiceRepository.java]: fixed!

**Pull request merged**